aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -387,6 +383,90 @@ impl AptosVM {
         }
     }
 
+    fn finish_aborted_transaction(
+        &self,
+        gas_meter: &mut impl AptosGasMeter,
+        txn_data: &TransactionMetadata,
+        resolver: &impl AptosMoveResolver,
+        log_context: &AdapterLogSchema,
+        change_set_configs: &ChangeSetConfigs,
+    ) -> Result<(VMChangeSet, FeeStatement), VMStatus> {
+        if txn_data.fee_payer().is_some()
+            && txn_data.sequence_number == 0
+            && self
+                .vm_impl
+                .get_features()
+                .is_enabled(FeatureFlag::SPONSORED_AUTOMATIC_ACCOUNT_CREATION)
+        {
+            // Clear side effects: create new session and clear refunds from fee statement.
+            let mut session = self
+                .vm_impl
+                .new_session(resolver, SessionId::run_on_abort(txn_data));
+
+            create_account_if_does_not_exist(&mut session, gas_meter, txn_data.sender())
+                .map_err(expect_no_verification_errors)
+                .or_else(|err| {
+                    expect_only_successful_execution(
+                        err,
+                        &format!("{:?}::{}", ACCOUNT_MODULE, CREATE_ACCOUNT_IF_DOES_NOT_EXIST),
+                        log_context,
+                    )
+                })?;
+
+            let mut change_set = session.finish(change_set_configs)?;
+            if let Err(err) = self.charge_change_set(&mut change_set, gas_meter, txn_data) {
+                info!(
+                    *log_context,
+                    "Failed during charge_change_set: {:?}. Most likely exceded gas limited.", err,
+                );
+            };
+
+            // zero out storage refund
+            let storage_refund = 0;
+            let fee_statement =
+                AptosVM::fee_statement_from_gas_meter(txn_data, gas_meter, storage_refund);
+
+            let session_id = SessionId::epilogue_meta(txn_data);
+            let mut respawned_session = RespawnedSession::spawn(
+                self,
+                session_id,
+                resolver,
+                change_set,
+                storage_refund.into(),
+            )?;
+            respawned_session.execute(|session| {
+                self.vm_impl.run_failure_epilogue(
+                    session,
+                    gas_meter.balance(),
+                    fee_statement,
+                    txn_data,
+                    log_context,
+                )
+            })?;
+            respawned_session
+                .finish(change_set_configs)
+                .map(|set| (set, fee_statement))
+        } else {
+            // Clear side effects: create new session and clear refunds from fee statement.
+            let mut session = self
+                .vm_impl
+                .new_session(resolver, SessionId::run_on_abort(txn_data));
+
+            let fee_statement = AptosVM::fee_statement_from_gas_meter(txn_data, gas_meter, 0);
```

Comment:
how about doing the same thing as you did at the other site (`let stoarge_fee = 0`) to make it easier to read. (or remove the blank line at least)

Reply:
sure